static/app/views/performance/mobile/usePlatformSelector.tsx

```diff
@@ -0,0 +1,45 @@
+import {useMemo} from 'react';
+
+import {decodeScalar} from 'sentry/utils/queryString';
+import {useLocation} from 'sentry/utils/useLocation';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import useProjects from 'sentry/utils/useProjects';
+import {PlatformSelector} from 'sentry/views/performance/mobile/screenload/screens/platformSelector';
+import {isCrossPlatform} from 'sentry/views/performance/mobile/screenload/screens/utils';
+
+export const PLATFORM_LOCAL_STORAGE_KEY = 'mobile-performance-platform';
+export const PLATFORM_QUERY_PARAM = 'platform';
+export const DEFAULT_PLATFORM = 'Android';
+
+function usePlatformSelector() {
+  const {selection} = usePageFilters();
+  const {projects} = useProjects();
+  const location = useLocation();
+
+  const project = useMemo(() => {
+    if (selection.projects.length !== 1) {
+      return null;
+    }
+
+    return projects.find(p => p.id === String(selection.projects));
+  }, [projects, selection.projects]);
+
+  const isProjectCrossPlatform = useMemo(
+    () => (project ? isCrossPlatform(project) : false),
+    [project]
+  );
+
+  const selectedPlatform =
+    decodeScalar(location.query[PLATFORM_QUERY_PARAM]) ??
+    localStorage.getItem(PLATFORM_LOCAL_STORAGE_KEY) ??
+    DEFAULT_PLATFORM;
+
+  return {
+    project,
+    selectedPlatform,
+    isProjectCrossPlatform,
+    PlatformSelector,
```

Comment:
question: how come we return `PlatformSelector` here? Doesn't seem like we make any changes to the component here, so I think we can make this hook lighter by expecting users to import this component on their own.

Reply:
Yeah, I thought this was a bit odd but I figured it was okay since this hook was also returning the selected platform, so it made sense to return the component that modified the selection. I looked it up and it's supposed to be pretty anti-pattern-y because it doesn't allow for flexibility so I took it out and renamed the hook to `useCrossPlatformProject`. It's not the best name but I'm unsure of what to call it for now